ts/private/ts_project_worker.js

```diff
@@ -463,111 +652,132 @@ function createProgram(args, initialInputs) {
         }
     }
 
+    function disableStatisticsAndTracing() {
+        ts.performance.disable();
+        if (!ts.tracing && ts.startTracing) {
+            ts.tracing.stopTracing()
+        }
+    }
+
+    function postRun() {
+        if (ts.performance && ts.performance.isEnabled()) {
+            ts.performance.forEachMeasure((name, duration) => write(`${name} time: ${duration}\n`));
+            ts.performance.disable()
+            ts.performance.enable()
+        }
+
+        if (ts.tracing) {
+            ts.tracing.stopTracing()
+        }
+    }
+
     function updateOutputs() {
         outputs.clear();
         if (compilerOptions.tsBuildInfoFile) {
-            const p = path.relative(execRoot, path.join(bin, compilerOptions.tsBuildInfoFile));
+            const p = path.relative(execroot, path.join(bin, compilerOptions.tsBuildInfoFile));
             outputs.add(p);
         }
     }
 
-    function checkAndApplyArgs(newArgs) {
+    function applySyntheticOutPaths() {
+        compilerOptions.outDir = path.join("__synthetic__outdir__", compilerOptions.outDir);
```

Comment:
nit: shorten `__synthetic__outdir__` incase you're looking at debug logs to reduce line length